node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -853,23 +667,21 @@ async fn handle_our_view_change(
 		state.span_per_relay_parent.remove(&removed);
 	}
 
-	state.active_paras.assign_incoming(ctx.sender(), keystore, added).await;
-	state.active_paras.remove_outgoing(removed);
-
-	for (peer_id, peer_data) in state.peer_data.iter_mut() {
-		peer_data.prune_old_advertisements(&state.view);
+	peer_slots::cycle_para(
+		ctx.sender(), &mut state.peer_slots.active_paras, keystore, added, removed,
+	).await;
 
+	for peer_id in peer_slots::handle_view_change(&mut state.peer_slots, &state.view).into_iter() {
 		// Disconnect peers who are not relevant to our current or next para.
 		//
 		// If the peer hasn't declared yet, they will be disconnected if they do not
 		// declare.
-		if let Some(para_id) = peer_data.collating_para() {
-			if !state.active_paras.is_current_or_next(para_id) {
-				disconnect_peer(ctx, peer_id.clone()).await;
-			}
-		}
+		disconnect_peer(ctx, peer_id).await;
 	}
 
+	state.evicted_peers.clear();
```

Comment:
why do we clear evicted peers on our view change which happens on every new relay block import?